enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/consensus/schedule/Timer.java

```diff
@@ -124,7 +124,9 @@ public synchronized void reset()
      * then a subsequent cancel will not ensure that the first execution of the
      * handler was cancelled.
      *
-     * @param mayInterruptIfRunning Interrupt any currently running job.
+     * N.B: The job will not be interrupted when await is used!
```

Comment:
So the pair of booleans select from three possible behaviours. Maybe replace them both with a three valued Enum?